### DISCUSSION TITLE
[react-markdown] How render html code from strapi

### DISCUSSION THREAD

**Discussion by dapzer** [remarkjs/remark#802] [Q&A] [answered]
How to make it so that I could put the html code in the strap and styles are attached to it

**Comment by ChristianMurphy** [comment, marked as answer]
@Dapzer could you share which library you are trying to use? `remark`, `react-markdown`, `rehype`, etc?
And how/where you want to render it? With React, Vue, something else? Client side, server side, etc?

Taking an educated guess, you probably want client side rendering with react, in which case https://github.com/remarkjs/react-markdown#appendix-a-html-in-markdown is likely what you are looking for.

**Reply by dapzer** [reply]
i use next js and react-markdown and I need the ability to post and render code from a strapi

**Reply by ChristianMurphy** [reply]
Use the content API to get your content https://strapi.io/documentation/developer-docs/latest/developer-resources/content-api/integrations/next-js.html#create-a-next-js-app and  https://strapi.io/documentation/developer-docs/latest/developer-resources/content-api/content-api.html#api-endpoints
Assign the content to state https://reactjs.org/docs/hooks-state.html
then pass the state, with the content to `react-markdown` https://github.com/remarkjs/react-markdown#use

:notebook: this will render client side, in the browser

---

If you want server side rendering see https://github.com/remarkjs/react-remark/issues/16 and https://github.com/remarkjs/react-remark#server-side-rendering

**Reply by dapzer** [reply]
![image](https://user-images.githubusercontent.com/78279257/128603981-d5d0603d-426e-4185-8a85-8784b31c630a.png)
you probably misunderstood me, I need to display the block shown in the screenshot and apply styles to it

**Reply by ChristianMurphy** [reply]
I don't understand what you are asking.
Could you share a codesandbox of what you've tried? (https://codesandbox.io/s/react-markdown-debug-9n4eg is a good starter)
Could you give more context on what you mean?

**Reply by dapzer** [reply]
Strapi 
![image](https://user-images.githubusercontent.com/78279257/128606769-5099a6f4-b3ed-4758-b91b-47cf511f7b1e.png)
What i get
![image](https://user-images.githubusercontent.com/78279257/128606849-2d86b97a-b666-41e0-8308-19b32973626f.png)
What i want
![image](https://user-images.githubusercontent.com/78279257/128606907-b9d42d52-53ee-48ff-a7a9-7b1d0ce43e61.png)

**Reply by ChristianMurphy** [reply]
Okay, and the part that makes it look how you want is `{styles.cdb}` and `{styles.cdbSub}`?

**Reply by dapzer** [reply]
yes

**Reply by ChristianMurphy** [reply]
That is JSX.
`react-markdown` is markdown, with the option to embed HTML.
If you want markdown with the option to embed JSX, you're looking for MDX (probably https://github.com/wooorm/xdm#next or https://github.com/wooorm/xdm#evaluatefile-options, alternatively with the original version https://github.com/mdx-js/mdx)

**Reply by dapzer** [reply]
Can i use mdx + react-markdown?

**Reply by ChristianMurphy** [reply]
It may be possible, but I'd recommend using xdm/mdx directly.
`evaluate` in xdm, and `Runtime` in the original mdx are roughly equivalent to `react-markdown` and are already configured to handle translating MDX to react.

**Reply by dapzer** [reply]
Thank you!

**Reply by ChristianMurphy** [reply]
Another alternative worth noting here.
Add the `class` name directly in the HTML, avoiding the JSX indirection.
`react-markdown` works fine with plain HTML, and with a consistent CSS naming convention, CSS-in-JS and CSSModules aren't really needed.

**Reply by dapzer** [reply]
Can you show me an example?

**Reply by ChristianMurphy** [reply]
https://codesandbox.io/s/react-markdown-debug-forked-tg9bi?file=/src/App.js

**Reply by dapzer** [reply]
I get this error, maybe it's because of the next?
![image](https://user-images.githubusercontent.com/78279257/128609191-66012682-9af7-426c-adc1-42385d9b0999.png)

**Reply by ChristianMurphy** [reply]
https://github.com/remarkjs/react-markdown/issues/626

**Reply by dapzer** [reply]
How change version?

**Reply by ChristianMurphy** [reply]
open `package.json`, find `rehype-react`, change the version number, run `npm install` (or `yarn`)

**Reply by dapzer** [reply]
does not work
![image](https://user-images.githubusercontent.com/78279257/128609559-3e4a5a04-06cf-427c-993a-e803ea5c43c5.png)
![image](https://user-images.githubusercontent.com/78279257/128609564-a08edefe-8e3c-4b40-86f0-9d3ecaf9d513.png)

**Reply by ChristianMurphy** [reply]
make sure to have a style sheet (with your styles) https://codesandbox.io/s/react-markdown-debug-forked-tg9bi?file=/src/App.css and include it in the component where you are using `react-markdown` https://codesandbox.io/s/react-markdown-debug-forked-tg9bi?file=/src/App.js:96-116

**Reply by dapzer** [reply]
Thank you very much, you helped me a lot, everything works as it should!
